### PR TITLE
tests: refine process exit path in failpoint test

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 
In some failpoint injection, we `panic` or `os.Exit(1)` directly, which will lose coverage statistics of the injection code

What we want to achieve is:
1. we have a way to inject error, and the code after injection must be not executed. This ensures the logical correctness.
2. the gotest coverage should collect the execute record of our injection code. This ensures coverage accuracy.

### What is changed and how it works?
1. run `main` function in a separate routine, currently apply change to `DM-worker.test` only
2. add a mocked `utils.OsExit` function to simulate running process terminate

### Check List 

Tests 

 - Integration test

### CODE REVIEW

File: cmd/dm-master/main_test.go [hunk 1]
```diff
@@ -32,6 +34,12 @@ func TestRunMain(t *testing.T) {
 		}
 	}
 
+	defer func() {
+		if err := recover(); err != nil {
```
Comment (csuzhangxc) [inline cmd/dm-master/main_test.go]: if we `panic` in a separate goroutine, this `recover` in the main goroutine cannot recover it.
Can this achieve our goals?

Comment (amyangfei) [reply cmd/dm-master/main_test.go]: yep, this still need to solve, I will dig into it later.

**Comment by GregoryIan** [comment]
any update?

File: cmd/dm-worker/main_test.go [hunk 2]
```diff
@@ -32,6 +40,25 @@ func TestRunMain(t *testing.T) {
...
+	go func() {
+		main()
+		close(waitCh)
+	}()
+
+	select {
+	case <-waitCh:
+		return
+	case <-exit:
+		return
```
Comment (csuzhangxc) [inline cmd/dm-worker/main_test.go]: so the method is to exit from this _main_ routine to terminal the separate `main()` routine (and all other routines)?

File: loader/loader.go [hunk 3]
```diff
@@ -141,7 +141,7 @@ func (w *Worker) run(ctx context.Context, fileJobQueue chan *fileJob, workerWg *
 					threshold, _ := val.(int)
 					if job.offset >= int64(threshold) {
 						log.Warnf("[failpoint] load offset %d execeeds threshold %d, will panic", job.offset, threshold)
-						panic("LoadExceedOffsetExit")
+						utils.OsExit(1)
```
Comment (csuzhangxc) [inline loader/loader.go]: can `utils.OsExit` carry some msg like `LoadExceedOffsetExit`?

File: cmd/dm-worker/main_test.go (hunk 2 above)
Comment (amyangfei) [reply cmd/dm-worker/main_test.go]: yes. (to achieve the goal-2, `main` routine must exit normally)

File: loader/loader.go (hunk 3 above)
Comment (amyangfei) [reply loader/loader.go]: The error message is not necessary, as we can log enough information before `utils.OsExit`, besides the function type `OsExit func(int)` is more natural to an `os.Exit` mock

**Comment by GregoryIan** [comment]
LGTM

**Pull request merged**